app/models/manageiq/providers/ibm_cloud/inventory/parser/power_virtual_servers.rb

```diff
@@ -248,10 +249,13 @@ def networks
 
   def sshkeys
     collector.sshkeys.each do |tkey|
+      cert = OpenSSL::X509::Certificate.new(tkey.ssh_key + "\n")
+      fingerprint = OpenSSL::Digest::SHA1.hexdigest(cert.to_der).scan(/../).join(':')
       tenant_key = {
         :creationDate => tkey.creation_date,
         :name         => tkey.name,
         :sshKey       => tkey.ssh_key,
+        :fingerprint  => fingerprint
       }
```

Comment:
@Kuldip-Nanda Where is this tenant_key actually used?  When it build the auth_key_pair below, it only uses the name, so the rest of these are not needed at all?

Reply:
I don't think tenant_key used anywhere.. possibly a past oversight?

Would be more clear to get rid of tenant_key altogether and just do:
```ruby
      persister.auth_key_pairs.build(           
        :name        => tkey.name,              
        :fingerprint => fingerprint,
        :public_key  => tkey.ssh_key
      )
```

Reply:
A spec can be added to ensure this is working too.

Reply:
Also, I think we need to set an `ems_ref` here. Since the keys are owned at the (PowerVS) tenant level, we end up with duplicate DB entries for ssh keys (1 x PowerVirtualServers::CloudManager even though they're shared). The PowerVS managers have access to the tenant_id [[ref]](https://github.com/ManageIQ/manageiq-providers-ibm_cloud/blob/master/app/models/manageiq/providers/ibm_cloud/power_virtual_servers/manager_mixin.rb#L49-L52), we just need something like `:ems_ref     => "#{pcloud_tenant_id}-#{tkey.name}"` here.

Reply:
@jaywcarman  thanks for pointing this out. Let me make this change.